### ISSUE TITLE
Add definitions for ambiguous tags in docs

### ISSUE THREAD

**Issue by eightnoneone** [corretto/corretto-docker#197]
The current list of variant tags for images use terms that are not defined well. 

Though some are terms of art that many Java developers and operators are familiar with, others are generic and ambiguous. The docker-library/docs [README.md](https://github.com/docker-library/docs/blob/master/amazoncorretto/README.md) has a section for variants that only addresses `alpine`.

As single point of reference, I am a non-developer ops engineer that has been deploying Java applications for 25 years. This is what I understand from the tags:

- `jre` - Just the runtime
- `jdk` - Everything needed for compile, and the runtime
- `full` - No idea how this is different from `jdk`
- `headless` - Removed dependencies of GUI human interface, but is it `jdk` or `jre`?
- `headful` - First use I've ever seen. Assumed opposite of headless, but how different from `full` or `jdk`?
- `generic` - No idea
- `native` - No idea
- `alpine` - Well defined in docs README ✅
- `alpine3` - I assume a newer version of alpine, but I'm not an alpine guy so I'm not sure of the impact.
- `al2` - AmazonLinux2, but consider that may not be obvious to people outside the Amazon sphere.
- `al2023` - AmazonLinux2023, same issue as `al2` and `alpine3`. How does this matter in my choice?

I would like to see more explicit definitions included in this repo's [README](https://github.com/corretto/corretto-docker/blob/main/README.md) and the docker-library/docs [README](https://github.com/docker-library/docs/blob/master/amazoncorretto/README.md).

**Comment by caojoshua** [comment]
Agreed that many of these labels are not obvious. We have this work planned internally. The updates will be in this repo's readme and the readme in docker-library

**Comment by keeganwitt** [comment]
https://github.com/docker-library/docs/blob/master/amazoncorretto/README.md#image-variants is out of date with this repo's README. Incremental progress can be made on this card by updating the official docs with what you have so far.
